samples/net/sockets/http_server/src/main.c

```diff
@@ -37,59 +44,139 @@ struct http_resource_detail_static index_html_gz_resource_detail = {
 	.static_data_len = sizeof(index_html_gz),
 };
 
-HTTP_RESOURCE_DEFINE(index_html_gz_resource, test_http_service, "/",
-		     &index_html_gz_resource_detail);
+static struct http_resource_detail_static main_js_gz_resource_detail = {
+	.common = {
+			.type = HTTP_RESOURCE_TYPE_STATIC,
+			.bitmask_of_supported_http_methods = BIT(HTTP_GET),
+			.content_encoding = "gzip",
+			.content_type = "text/javascript",
+		},
+	.static_data = main_js_gz,
+	.static_data_len = sizeof(main_js_gz),
+};
 
-static uint8_t recv_buffer[1024];
+static int uptime_handler(struct http_client_ctx *client, enum http_data_status status,
+			  uint8_t *buffer, size_t len, void *user_data)
+{
+	static bool response_sent;
+
+	LOG_DBG("Uptime handler status %d, response_sent %d", status, response_sent);
 
-static int dyn_handler(struct http_client_ctx *client, enum http_data_status status,
+	switch (status) {
+	case HTTP_SERVER_DATA_ABORTED:
+	{
+		response_sent = false;
+		return 0;
+	}
+
+	case HTTP_SERVER_DATA_MORE:
+	{
+		/* A payload is not expected with the GET request. Ignore any data and wait until
+		 * final callback before sending response
+		 */
+		return 0;
+	}
+
+	case HTTP_SERVER_DATA_FINAL:
+	{
+		if (response_sent) {
+			/* Response already sent, return 0 to indicate to server that the callback
+			 * does not need to be called again.
+			 */
+			response_sent = false;
+			return 0;
+		}
+
+		response_sent = true;
+		return snprintf(buffer, sizeof(uptime_buf), "%" PRId64, k_uptime_get());
+	}
+	default:
+	{
+		LOG_WRN("Unexpected status %d", status);
+		return -1;
+	}
+	}
+}
+
+static struct http_resource_detail_dynamic uptime_resource_detail = {
+	.common = {
+			.type = HTTP_RESOURCE_TYPE_DYNAMIC,
+			.bitmask_of_supported_http_methods = BIT(HTTP_GET),
+		},
+	.cb = uptime_handler,
+	.data_buffer = uptime_buf,
+	.data_buffer_len = sizeof(uptime_buf),
+	.user_data = NULL,
+};
+
+static void led_control(bool on)
+{
+	LOG_INF("POST request turning LED %s", on ? "on" : "off");
+
+	if (leds_dev != NULL) {
+		if (on) {
+			led_on(leds_dev, 0);
+		} else {
+			led_off(leds_dev, 0);
+		}
+	}
+}
+
+static void parse_led_post(uint8_t *buf, size_t len)
+{
+	if ((len == sizeof("led=on") - 1) && (0 == memcmp("led=on", buf, len))) {
+		led_control(true);
+	} else if ((len == sizeof("led=off") - 1) && (0 == memcmp("led=off", buf, len))) {
+		led_control(false);
+	} else {
+		LOG_WRN("Unexpected POST payload");
+	}
+}
```

Comment:
I feel like it would be kind of neat to use this as an excuse to show a JSON payload being parsed, but also OK to keep as-is

related, @rlubos @jukkar the sample enables `CONFIG_JSON_LIBRARY` but it looks like maybe it's a leftover since I don't think it's needed (yet! 😄)

Reply:
Perhaps a payload like:
```
{
  led_num: 0,
  led_state: true,
}
```
then we can keep the web interface the same (just always set `led_num` to 0), but if you send a POST via `curl` or similar you can control multiple LEDs (if they exist on the board)?

Shouldn't be much more complicated to implement than what I've done already.

Reply:
> the sample enables `CONFIG_JSON_LIBRARY` but it looks like maybe it's a leftover since I don't think it's needed

Yes, that is a historical artifact that is indeed not needed atm.

Reply:
JSON payload added for the POST request now.